File: python/exercise_205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hashmap = {}
        hashmap2 = {}
        for i in range(len(s)):
            if s[i] in hashmap:
                if t[i] != hashmap[s[i]]:
                    return False
            else:
                hashmap[s[i]] = t[i]
            if t[i] in hashmap2:
                if s[i] != hashmap2[t[i]]:
                    return False
            else:
                hashmap2[t[i]] = s[i]
        return True

File: java/hashsetOrHashmap/exercise_205.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character,Character> StoT = new HashMap<>();
        HashMap<Character,Character> TtoS = new HashMap<>();
        for(int i=0;i<s.length();i++){
            char sCurrent = s.charAt(i);
            char tCurrent = t.charAt(i);
            if(StoT.containsKey(sCurrent)){
                if(TtoS.containsKey(tCurrent)){
                    if(TtoS.get(tCurrent)!=sCurrent || StoT.get(sCurrent)!=tCurrent){
                        return false;
                    }
                }else{
                    return false;
                }
            }else{
                if(TtoS.containsKey(tCurrent)){
                    return false;
                }else{
                    StoT.put(sCurrent,tCurrent);
                    TtoS.put(tCurrent,sCurrent);
                }
            }
        }
        return true;
    }
}